src/uct/ib/dc/verbs/dc_verbs.c

```diff
@@ -573,8 +573,9 @@ ucs_status_t uct_dc_verbs_iface_create_ah(uct_dc_iface_t *dc_iface, uint16_t lid
     ah_attr.port_num      = iface->config.port_num;
     ah_attr.static_rate   = 0;
 
-    if ((iface->addr_type != UCT_IB_ADDRESS_TYPE_GLOBAL) &&
-        (iface->gid.global.subnet_prefix == gid->global.subnet_prefix)) {
+    if ((gid == NULL) ||
+       ((iface->addr_type != UCT_IB_ADDRESS_TYPE_GLOBAL) &&
```

Comment:
it means that if the iface address type is "local", and it gets FC message from a remote peer which contains valid GID, it would still use a "local" address, right?

Reply:
right, gid pointer will be NULL in case if peer requested local interaction